src/EditorFeatures/CSharpTest/Completion/CompletionProviders/AwaitCompletionProviderTests.cs

```diff
@@ -0,0 +1,283 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Completion.Providers;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionProviders;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Recommendations
+{
+    /// <summary>
+    /// The <see cref="AwaitCompletionProvider"/> adds async modifier if the return type is Task or ValueTask.
+    /// The tests here are only checking whether the completion item is provided or not.
+    /// Tests for checking adding async modifier are in:
+    /// src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_AwaitCompletion.vb
+    /// </summary>
+    [Trait(Traits.Feature, Traits.Features.Completion)]
+    public class AwaitCompletionProviderTests : AbstractCSharpCompletionProviderTests
+    {
+        internal override Type GetCompletionProviderType() => typeof(AwaitCompletionProvider);
+
+        private async Task VerifyAbsenceAsync(string code)
+        {
+            await VerifyItemIsAbsentAsync(code, "await");
+        }
+
+        private async Task VerifyAbsenceAsync(string code, LanguageVersion languageVersion)
```

Comment:
i'm surprised there is any langauge version specific code in this feature area.

Reply:
@CyrusNajmabadi This was a direct port from the deleted test file. The language version is used for top-level statement tests. See the current `AwaitKeywordRecommenderTests` in current main (this is being deleted in this PR - or essentially renamed to `AwaitCompletionProviderTests` with some refactoring you and new tests (in addition to the integration tests added)